### PR TITLE
drachodran-e22a348 Assignment 4 Hand-in

### PR DESCRIPTION
**Pull request by drachodran**

### CODE REVIEW

File: Assignment4/drachodran-e22a348/task06.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1y93Mcblef-xExVKjumM-Xl1wyT1659mG
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/drachodran-e22a348/task06.py:12]: wont work in python. This is notebook magic

File: Assignment4/drachodran-e22a348/task07.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      ?individuals rdf:type ?subclasses.
+      ?individuals ?properties ?o.
+    }
+  }
+  ''', initNs = { "ns": ns, "rdf": RDF, "rdfs": RDFS }
+  )
+for y in g.query(q3):
+  print(y)
+
+# Visualize the results
```
Comment (dgarijo) [inline Assignment4/drachodran-e22a348/task07.py:89]: missing exercises in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]

File: Assignment4/drachodran-e22a348/task06.py (hunk 1 above)
Comment (drachodran) [reply Assignment4/drachodran-e22a348/task06.py:12]: Hi Daniel! Could you please tell me more about the problem you see? Do you mean my exercise works on Google colab but it wouldn't execute on a local Python interpreter? As I am new to Python, I can't see the problem

Comment (dgarijo) [reply Assignment4/drachodran-e22a348/task06.py:12]: line 12 is a magic command (starts with !). Therefore it won't work in Python, it's for the notebooks/bash scripts

**Comment by dgarijo** [comment]
Exercises 7.1 .. 7.3 are still missing in rdflib

File: Assignment4/drachodran-e22a348/task06.py (hunk 1 above)
Comment (drachodran) [reply Assignment4/drachodran-e22a348/task06.py]: Dear Daniel, please check that I unmagicked the files. It should be accepted now, I hope.

File: Assignment4/drachodran-e22a348/task07.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      ?subclasses rdfs:subClassOf* ns:Person.
+      ?individuals rdf:type ?subclasses.
+      ?individuals ?properties ?o.
+    }
+  }
+  ''', initNs = { "ns": ns, "rdf": RDF, "rdfs": RDFS }
+  )
+for y in g.query(q3):
+  print(y)
+
```
Comment (dgarijo) [inline Assignment4/drachodran-e22a348/task07.py:88]: You are still missing the exercises 7.1..7.3 using rdf lib. They are solved for SPARQL only

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please complete missing exercises

File: Assignment4/drachodran-e22a348/task07.py (hunk 3 above)
Comment (drachodran) [reply Assignment4/drachodran-e22a348/task07.py:101]: Hi Dani,

it should be OK now.

**Review by dgarijo — APPROVED** [review]

**Pull request merged**